### DISCUSSION TITLE
ACP-108: EVM Event Importing

### DISCUSSION THREAD

**Discussion by avalanche-foundation-admin** [avalanche-foundation/ACPs#114] [Discussions]
Proposes a standard smart contract interface and abstract implementation for importing EVM events from any blockchain within Avalanche using [Avalanche Warp Messaging](https://docs.avax.network/build/cross-chain/awm/overview). Allows for proving events that have occurred on other chains without interaction on those chains.

https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/108-evm-event-importing

**Comment by michaelkaplan13** [comment]
The two possible use cases for this type of event importing that first pop into my mind are:

- Importing data feeds into Subnets that are published on other chains (i.e. price feeds currently on the C-Chain)
- Proving that an NFT on another chain was minted/transferred to an address. Could be used to grant special permissions to those account in games, special discounts to those accounts for services, etc.

For anyone developing (or planning to) using Subnets, I'd be curious to hear thoughts on the first as relates to the question of time delay between event publishing and event importing as outlined [here](https://github.com/ava-labs/event-importer-poc?tab=readme-ov-file#is-the-delay-from-the-time-an-event-is-emitted-on-a-source-chain-to-when-it-is-imported-on-another-chain-a-non-starter-for-certain-applications).

**Reply by FrancoisPaul69** [reply]
Hello Michael,

May I have your opinion on the ACP and comment below : 
https://github.com/avalanche-foundation/ACPs/discussions/78#discussioncomment-10563732

Regards,
François

**Comment by meeshhhh** [comment]
Expanding on use cases... here are some more ideas, would love to get people's thoughts!

- **NFT Ownership + Identity**: Michael's idea above, where an NFT minted on Chain A can unlock special features or content on Chain B. Could be used in the gaming industry where completing a quest on one chain unlocks items, skills, etc. on another chain. Could also be used for identity/compliance purposes for institutional chains - e.g. "identity chain" holds KYC’d information about users, this info can be imported to any other chain (JPM, MS, GS) and users will have access to transacting on these chains.
- **Compliance/Auditing**: Create a “compliance chain” that audits and verifies txs that occurred on the chain being audited. Useful for banks and financial institutions that may want to create efficient (fast) and accountable (immutable) internal or external, audit chains. 
- **Supply Management**: Similar to the accountability with a “compliance chain”, could create a chain that verifies supply chain events across multiple chains to ensure the integrity and traceability of goods. Useful for tracking ESG for retail companies that produce clothing, food, and other tangible goods. 
- **Cross-chain Governance**: Voting or governance on one chain could be imported and enacted on another chain. Could have one DAO chain that facilitates governance across multiple ecosystems. 
- **Restaking**: Stake AVAX on C-Chain and opt in to restake with a different protocol on Chain B, automatically earning more rewards on Chain B without interacting directly with C-Chain. 
- **Boosting User Engagement**: Incentivize certain actions on one chain by rewarding/airdropping users automatically on a different chain (e.g. C-Chain).

**Comment by meeshhhh** [comment]
Please see [here](https://github.com/avalanche-foundation/ACPs/discussions/120) to join the upcoming Developer Community Call on 7-16-2024 for ACP-108. The discussion will focus on highly anticipated use cases for EVM Event Importing.
